edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Metric.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Metric
+    {
+        public DateTime TimeGeneratedUtc { get; }
+        public string Name { get; }
+        public double Value { get; }
+        public string Tags { get; }
+
+        public Metric(DateTime timeGeneratedUtc, string name, double value, string tags)
+        {
+            Preconditions.CheckArgument(timeGeneratedUtc.Kind == DateTimeKind.Utc, $"Metric {nameof(timeGeneratedUtc)} parameter only supports in UTC.");
+            this.TimeGeneratedUtc = timeGeneratedUtc;
+            this.Name = Preconditions.CheckNotNull(name, nameof(name));
+            this.Value = value;
+            this.Tags = Preconditions.CheckNotNull(tags, nameof(tags));
+        }
+
+        /// <summary>
+        /// This combines the hashes of name and tags to make it easy to group all metrics.
+        /// </summary>
+        /// <returns>Hash of name and tags.</returns>
+        public int GetMetricKey()
+        {
+            // TODO: replace with "return HashCode.Combine(Name.GetHashCode(), Tags.GetHashCode());"
```

Comment:
Missing unchecked statement.

public override int GetHashCode()
        {
            unchecked
            {
                int hashCode = this.Name.GetHashCode();
                hashCode = (hashCode * 397) ^ this.Tags.GetHashCode();
                return hashCode;
            }
        }

Reply:
unchecked is the default behavior for addition. I'm using a slightly different method of combining hashes, so there's no ^ in mine.